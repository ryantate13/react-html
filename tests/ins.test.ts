import {ins} from '../src';

describe('ins', () => {
    it('renders with children', () => {
        expect(ins({}, 'TEST')).toMatchSnapshot();
    });
    
    it('renders arrays of children', () => {
        expect(ins({}, Array(10).fill('TEST'))).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(ins({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(ins()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(ins({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(ins({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(ins({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(ins({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(ins({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(ins({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(ins({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(ins({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(ins({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(ins({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(ins({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(ins({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(ins({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(ins({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(ins({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden=true', () => {
        expect(ins({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden=false', () => {
        expect(ins({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(ins({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(ins({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(ins({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(ins({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(ins({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(ins({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(ins({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(ins({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(ins({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(ins({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(ins({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(ins({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(ins({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope=true', () => {
        expect(ins({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope=false', () => {
        expect(ins({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(ins({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(ins({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(ins({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(ins({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(ins({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(ins({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style={"color":"#000000"}', () => {
        expect(ins({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex=1', () => {
        expect(ins({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(ins({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(ins({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(ins({translate: "no"})).toMatchSnapshot();
    });

    it('renders with cite="https://google.com/"', () => {
        expect(ins({cite: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with dateTime="12:00"', () => {
        expect(ins({dateTime: "12:00"})).toMatchSnapshot();
    });
});