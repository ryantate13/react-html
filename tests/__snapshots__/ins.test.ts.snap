// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ins renders with accessKey="a,b,c,d" 1`] = `
<ins
  accessKey="a,b,c,d"
/>
`;

exports[`ins renders with autoCapitalize="characters" 1`] = `
<ins
  autoCapitalize="characters"
/>
`;

exports[`ins renders with autoCapitalize="none" 1`] = `
<ins
  autoCapitalize="none"
/>
`;

exports[`ins renders with autoCapitalize="off" 1`] = `
<ins
  autoCapitalize="off"
/>
`;

exports[`ins renders with autoCapitalize="on" 1`] = `
<ins
  autoCapitalize="on"
/>
`;

exports[`ins renders with autoCapitalize="sentences" 1`] = `
<ins
  autoCapitalize="sentences"
/>
`;

exports[`ins renders with autoCapitalize="words" 1`] = `
<ins
  autoCapitalize="words"
/>
`;

exports[`ins renders with children 1`] = `
<ins>
  TEST
</ins>
`;

exports[`ins renders with cite="https://google.com/" 1`] = `
<ins
  cite="https://google.com/"
/>
`;

exports[`ins renders with className="foo bar baz" 1`] = `
<ins
  className="foo bar baz"
/>
`;

exports[`ins renders with contentEditable="false" 1`] = `
<ins
  contentEditable="false"
/>
`;

exports[`ins renders with contentEditable="true" 1`] = `
<ins
  contentEditable="true"
/>
`;

exports[`ins renders with dateTime="12:00" 1`] = `
<ins
  dateTime="12:00"
/>
`;

exports[`ins renders with dir="auto" 1`] = `
<ins
  dir="auto"
/>
`;

exports[`ins renders with dir="ltr" 1`] = `
<ins
  dir="ltr"
/>
`;

exports[`ins renders with dir="rtl" 1`] = `
<ins
  dir="rtl"
/>
`;

exports[`ins renders with draggable="false" 1`] = `
<ins
  draggable="false"
/>
`;

exports[`ins renders with draggable="true" 1`] = `
<ins
  draggable="true"
/>
`;

exports[`ins renders with empty attributes 1`] = `<ins />`;

exports[`ins renders with hidden="false" 1`] = `
<ins
  hidden={false}
/>
`;

exports[`ins renders with hidden="true" 1`] = `
<ins
  hidden={true}
/>
`;

exports[`ins renders with id="TEST" 1`] = `
<ins
  id="TEST"
/>
`;

exports[`ins renders with inputMode="decimal" 1`] = `
<ins
  inputMode="decimal"
/>
`;

exports[`ins renders with inputMode="email" 1`] = `
<ins
  inputMode="email"
/>
`;

exports[`ins renders with inputMode="none" 1`] = `
<ins
  inputMode="none"
/>
`;

exports[`ins renders with inputMode="numeric" 1`] = `
<ins
  inputMode="numeric"
/>
`;

exports[`ins renders with inputMode="search" 1`] = `
<ins
  inputMode="search"
/>
`;

exports[`ins renders with inputMode="tel" 1`] = `
<ins
  inputMode="tel"
/>
`;

exports[`ins renders with inputMode="text" 1`] = `
<ins
  inputMode="text"
/>
`;

exports[`ins renders with inputMode="url" 1`] = `
<ins
  inputMode="url"
/>
`;

exports[`ins renders with is="foo" 1`] = `
<ins
  is="foo"
/>
`;

exports[`ins renders with itemID="https://google.com/" 1`] = `
<ins
  itemID="https://google.com/"
/>
`;

exports[`ins renders with itemProp="https://google.com/" 1`] = `
<ins
  itemProp="https://google.com/"
/>
`;

exports[`ins renders with itemRef="foo bar" 1`] = `
<ins
  itemRef="foo bar"
/>
`;

exports[`ins renders with itemScope="false" 1`] = `
<ins
  itemScope={false}
/>
`;

exports[`ins renders with itemScope="true" 1`] = `
<ins
  itemScope={true}
/>
`;

exports[`ins renders with itemType="https://google.com/" 1`] = `
<ins
  itemType="https://google.com/"
/>
`;

exports[`ins renders with lang="" 1`] = `
<ins
  lang=""
/>
`;

exports[`ins renders with lang="es-419" 1`] = `
<ins
  lang="es-419"
/>
`;

exports[`ins renders with slot="TEST" 1`] = `
<ins
  slot="TEST"
/>
`;

exports[`ins renders with spellCheck="false" 1`] = `
<ins
  spellCheck="false"
/>
`;

exports[`ins renders with spellCheck="true" 1`] = `
<ins
  spellCheck="true"
/>
`;

exports[`ins renders with style="[object Object]" 1`] = `
<ins
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`ins renders with tabIndex="1" 1`] = `
<ins
  tabIndex={1}
/>
`;

exports[`ins renders with title="TEST" 1`] = `
<ins
  title="TEST"
/>
`;

exports[`ins renders with translate="no" 1`] = `
<ins
  translate="no"
/>
`;

exports[`ins renders with translate="yes" 1`] = `
<ins
  translate="yes"
/>
`;

exports[`ins renders without attributes 1`] = `<ins />`;
